### ISSUE TITLE
[Bug] error train yolov3-lite 

### ISSUE THREAD

**Issue by shuxiao9058** [eric612/MobileNet-YOLO#117]
error train yolov3-lite 
```
tailf nohup.out
I0523 17:42:36.838488 15887 net.cpp:202] conv0/scale does not need backward computation.
I0523 17:42:36.838492 15887 net.cpp:202] conv0/bn does not need backward computation.
I0523 17:42:36.838496 15887 net.cpp:202] conv0 does not need backward computation.
I0523 17:42:36.838500 15887 net.cpp:202] data does not need backward computation.
I0523 17:42:36.838505 15887 net.cpp:244] This network produces output detection_eval
I0523 17:42:36.838567 15887 net.cpp:257] Network initialization done.
I0523 17:42:36.838929 15887 solver.cpp:70] Solver scaffolding done.
I0523 17:42:36.844560 15887 caffe.cpp:255] Starting Optimization
I0523 17:42:36.844571 15887 solver.cpp:302] Solving MobileNet-YOLO
I0523 17:42:36.844575 15887 solver.cpp:303] Learning Rate Policy: multistep
I0523 17:42:41.597106 15887 solver.cpp:252] Iteration 0 (-5.799e-34 iter/s, 4.75242s/10 iters), loss = 644.218
I0523 17:42:41.597175 15887 solver.cpp:271]     Train net output #0: det_loss1 = 201.067 (* 1 = 201.067 loss)
I0523 17:42:41.597190 15887 solver.cpp:271]     Train net output #1: det_loss2 = 444.498 (* 1 = 444.498 loss)
I0523 17:42:41.597221 15887 sgd_solver.cpp:121] Iteration 0, lr = 0.0005
I0523 17:42:44.900668 15887 yolov3_layer.cpp:411] noobj: 0.691635 obj: 0.454663 iou: 0.242478 cat: 0.721049 recall: 0.0889683 recall75: 0.0433333 count: 6
I0523 17:42:44.933354 15887 yolov3_layer.cpp:411] noobj: 0.488032 obj: 0.355171 iou: 0.179425 cat: 0.543738 recall: 0.03 recall75: 0 count: 3
I0523 17:42:52.004562 15887 yolov3_layer.cpp:411] noobj: 0.212177 obj: 0.217743 iou: 0.368333 cat: 0.927393 recall: 0.233571 recall75: 0.0166667 count: 5
I0523 17:42:52.031620 15887 yolov3_layer.cpp:411] noobj: 0.120318 obj: 0.0633717 iou: 0.281912 cat: 0.899958 recall: 0.173333 recall75: 0.0333333 count: 3
I0523 17:43:00.964931 15887 yolov3_layer.cpp:411] noobj: 0.0915175 obj: 0.0660229 iou: 0.377092 cat: 0.947308 recall: 0.203889 recall75: 0.0111111 count: 5
I0523 17:43:00.987834 15887 yolov3_layer.cpp:411] noobj: 0.0563753 obj: 0.021183 iou: 0.324921 cat: 0.947733 recall: 0.108974 recall75: 0 count: 3
I0523 17:43:09.124073 15887 yolov3_layer.cpp:411] noobj: 0.0623678 obj: 0.0386717 iou: 0.374178 cat: 0.967031 recall: 0.256164 recall75: 0.0268519 count: 4
I0523 17:43:09.141932 15887 yolov3_layer.cpp:411] noobj: 0.0382703 obj: 0.0206841 iou: 0.379594 cat: 0.954589 recall: 0.231633 recall75: 0.0119048 count: 4
I0523 17:43:17.771657 15887 yolov3_layer.cpp:411] noobj: 0.0493647 obj: 0.0338927 iou: 0.460246 cat: 0.974456 recall: 0.373968 recall75: 0.0857143 count: 5
I0523 17:43:17.792003 15887 yolov3_layer.cpp:411] noobj: 0.029333 obj: 0.0174566 iou: 0.324532 cat: 0.96801 recall: 0.126296 recall75: 0 count: 3
I0523 17:43:26.537012 15887 yolov3_layer.cpp:411] noobj: 0.0409369 obj: 0.0307595 iou: 0.420163 cat: 0.978666 recall: 0.270926 recall75: 0.00740741 count: 6
I0523 17:43:26.563143 15887 yolov3_layer.cpp:411] noobj: 0.0246718 obj: 0.0118164 iou: 0.350258 cat: 0.972484 recall: 0.0916667 recall75: 0.0142857 count: 3
I0523 17:43:28.980520 15887 solver.cpp:252] Iteration 10 (0.211047 iter/s, 47.3827s/10 iters), loss = 31.4523
I0523 17:43:28.980566 15887 solver.cpp:271]     Train net output #0: det_loss1 = 2.33867 (* 1 = 2.33867 loss)
I0523 17:43:28.980587 15887 solver.cpp:271]     Train net output #1: det_loss2 = 2.20879 (* 1 = 2.20879 loss)
I0523 17:43:28.980615 15887 sgd_solver.cpp:121] Iteration 10, lr = 0.0005
I0523 17:43:37.156095 15887 yolov3_layer.cpp:411] noobj: 0.0351622 obj: 0.0268283 iou: 0.423237 cat: 0.983361 recall: 0.331296 recall75: 0.0305556 count: 5
I0523 17:43:37.193518 15887 yolov3_layer.cpp:411] noobj: 0.0210809 obj: 0.00977233 iou: 0.378337 cat: 0.980884 recall: 0.188571 recall75: 0 count: 3
I0523 17:43:49.409251 15887 yolov3_layer.cpp:411] noobj: 0.0294884 obj: 0.0217133 iou: 0.420166 cat: 0.984678 recall: 0.335212 recall75: 0.0585185 count: 5
I0523 17:43:49.439169 15887 yolov3_layer.cpp:411] noobj: 0.0179037 obj: 0.0103777 iou: 0.394662 cat: 0.978646 recall: 0.263248 recall75: 0.025641 count: 3
I0523 17:43:57.636901 15887 yolov3_layer.cpp:411] noobj: 0.0255653 obj: 0.0188726 iou: 0.440703 cat: 0.987522 recall: 0.365291 recall75: 0.00740741 count: 6
I0523 17:43:57.664705 15887 yolov3_layer.cpp:411] noobj: 0.0156747 obj: 0.00953031 iou: 0.396581 cat: 0.982916 recall: 0.317778 recall75: 0.0444444 count: 3
I0523 17:44:07.858583 15887 yolov3_layer.cpp:411] noobj: 0.0233368 obj: 0.0166474 iou: 0.405767 cat: 0.98831 recall: 0.297355 recall75: 0.0355556 count: 6
I0523 17:44:07.895442 15887 yolov3_layer.cpp:411] noobj: 0.0141616 obj: 0.00640049 iou: 0.386442 cat: 0.984826 recall: 0.279762 recall75: 0.0714286 count: 2
I0523 17:44:21.365948 15887 yolov3_layer.cpp:411] noobj: 0.0207184 obj: 0.0159103 iou: 0.427837 cat: 0.989402 recall: 0.359127 recall75: 0.0361111 count: 6
I0523 17:44:21.384840 15887 yolov3_layer.cpp:411] noobj: 0.0125481 obj: 0.00839939 iou: 0.37792 cat: 0.98161 recall: 0.252564 recall75: 0.0384615 count: 2
I0523 17:44:28.980269 15887 solver.cpp:252] Iteration 20 (0.16667 iter/s, 59.9989s/10 iters), loss = 5.56917
I0523 17:44:28.980331 15887 solver.cpp:271]     Train net output #0: det_loss1 = 2.55332 (* 1 = 2.55332 loss)
I0523 17:44:28.980350 15887 solver.cpp:271]     Train net output #1: det_loss2 = 1.4091 (* 1 = 1.4091 loss)
I0523 17:44:28.980360 15887 sgd_solver.cpp:121] Iteration 20, lr = 0.0005
I0523 17:44:30.562855 15887 yolov3_layer.cpp:411] noobj: 0.0186681 obj: 0.0143026 iou: 0.420149 cat: 0.989232 recall: 0.302011 recall75: 0.0244444 count: 5
I0523 17:44:30.599351 15887 yolov3_layer.cpp:411] noobj: 0.011311 obj: 0.00694754 iou: 0.367632 cat: 0.986834 recall: 0.191667 recall75: 0.0222222 count: 4
I0523 17:44:37.184065 15887 blocking_queue.cpp:50] Waiting for data
```

**Comment by shuxiao9058** [comment]
Waiting for data

**Comment by shuxiao9058** [comment]
请问为什么会卡在“Waiting for data”这一步，怎么解决！ @eric612

**Comment by shuxiao9058** [comment]
网上查了一些资料说是io的问题

http://caffecn.cn/?/question/374
https://stackoverflow.com/questions/38424121/how-to-speed-up-image-loading-in-caffe
https://github.com/BVLC/caffe/issues/3264

解决方法好像修改data_param的prefetch就可以了，目前将该值设置到10正常。继续观察！

**Comment by shuxiao9058** [comment]
anyone can help me?

**Comment by eric612** [comment]
你的train跟test的lmdb是指向同一個檔案嗎

**Comment by shuxiao9058** [comment]
不是同一个数据集，一个是train数据集，一个是test数据集啊！另外网上查好些都说是io问题！但是我这边训练直接卡在`Waiting for data`这一步了！

**Comment by lq0104** [comment]
@eric612 如果train和test的lmdb指向同一路径会有问题吗？我之前这么做过，结果一算test yolo loss的时候就卡住不动了，是这样吗？

**Comment by shuxiao9058** [comment]
@lq0104 你遇到过这个问题吗？一直卡在这里训练不下去！

**Comment by lq0104** [comment]
哦，我的是这样：一进入到test_interval时，就会卡住不动，输出控制台上显示在计算yolo loss

**Comment by lq0104** [comment]
> @eric612 如果train和test的lmdb指向同一路径会有问题吗？我之前这么做过，结果一算test yolo loss的时候就卡住不动了，是这样吗？

但我刚才想了想，以上的描述有误，我的train和test的lmdb并没有指向同一个路径，只不过生成train和test的lmdb是从同一个数据集里转出来的

**Comment by shuxiao9058** [comment]
我的是卡在`Waiting for data`这一步不动了！

**Comment by lq0104** [comment]
我的不会呢，这步没问题啊，你是偶尔卡住还是一直卡住呢

**Comment by shuxiao9058** [comment]
是一直卡住！我用MobileNet-SSD则没有这个问题！

**Comment by lq0104** [comment]
猜测一下，batch_size设置的太大了？你的GPU显存使用情况如何啊

**Comment by lq0104** [comment]
#88 可以参考一下？或者你再搜搜相关问题关键字

**Comment by shuxiao9058** [comment]
@lq0104 我试一下，多谢！

**Comment by shuxiao9058** [comment]
> 猜测一下，batch_size设置的太大了？你的GPU显存使用情况如何啊

还是有问题，显存占用不高！另外，batch_size设置的是2依然有问题。

**Comment by shuxiao9058** [comment]
换了一台电脑，训练没有问题！
